src/optimise/train.jl

```diff
@@ -82,11 +82,16 @@ batchmemaybe(x::Tuple) = x
 
 """
     train!(loss, params, data, opt; cb)
-
+        
+`train!` uses a loss function (`loss`) and training data (`data`) to improve the 
+parameters (`params`) of your model based on a pluggable optimiser (`opt`).
+        
 For each datapoint `d` in `data`, compute the gradient of  `loss` with
```

Comment:
```suggestion
For each datapoint `d` in the `data` iterator, compute the gradient of  `loss` with
```
I don't love this phrasing, but barring a larger docstring refactor it seems the easiest way to point out what interfaces `data` should support.

Reply:
Do we need to say "iterator", isn't it assumed that any time we do `for something in something else`, `something else` is an iterator / iterable?

Reply:
I'm not sure, but my inclination would be that "in" is sufficiently general verbiage that new users wouldn't pattern match it to iteration. We can always leave this out and revisit if people aren't getting it.